### PR TITLE
[CI] Update GH Actions to test mne-stable

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Addresses: https://github.com/mne-tools/mne-python/issues/10609#issuecomment-1120222708

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-connectivity/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -132,6 +136,12 @@ jobs:
           git clone --depth 1 https://github.com/mne-tools/mne-python.git -b main
           pip install -e ./mne-python
 
+      - name: Install MNE (stable)
+        if: "matrix.mne-version == 'mne-stable'"
+        run: |
+          git clone --depth 1 https://github.com/mne-tools/mne-python.git -b maint/0.24
```
Comment (mscheltienne) [inline .github/workflows/unit_tests.yml:142]:
```suggestion
          git clone --depth 1 https://github.com/mne-tools/mne-python.git -b maint/1.0
```
Stable is 1.0?

**Comment by agramfort** [comment]
stable is what you install with pip using code on pypi

**Comment by mscheltienne** [comment]
Yes, so 1.0.2. I'm guessing you copy/pasted from `mne-connectivity`, so maybe you also want to update this workflow:

https://github.com/mne-tools/mne-connectivity/blob/2d47ae7d281c2ff2fe17f2d55f2e3d7dd0d0a4c5/.github/workflows/unit_tests.yml#L147

**Comment by hoechenberger** [comment]
I wouldn't install stable from the `git` repo as you'd have to keep updating the branch name. I'd just install it from PyPI…

**Comment by hoechenberger** [comment]
@adam2392 Could you please ping me once this passes & is merged? Then I'll go ahead and make a `conda-forge` package

File: mne_icalabel/iclabel/tests/test_label_components.py [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import pytest
+from pathlib import Path
 from mne.datasets import sample
 from mne.io import read_raw
 from mne.preprocessing import ICA
 
 from mne_icalabel.iclabel import iclabel_label_components
 
-directory = sample.data_path() / "MEG" / "sample"
+directory = Path(sample.data_path()) / "MEG" / "sample"
```
Comment (mscheltienne) [inline mne_icalabel/iclabel/tests/test_label_components.py:9]: `.data_path()` should already return Path

Comment (adam2392) [reply mne_icalabel/iclabel/tests/test_label_components.py:10]: Ah yes. you're right. this was only needed for v0.24-

**Comment by adam2392** [comment]
@hoechenberger this is done now and works with men-stable

File: mne_icalabel/iclabel/tests/test_label_components.py (hunk 2 above)
Comment (hoechenberger) [reply mne_icalabel/iclabel/tests/test_label_components.py]: I think you forgot to revert this / change it according to @mscheltienne's proposal

Comment (adam2392) [reply mne_icalabel/iclabel/tests/test_label_components.py]: I did! The diff is outdated here :p.

Comment (hoechenberger) [reply mne_icalabel/iclabel/tests/test_label_components.py]: Sorry, my bad!

**Pull request merged**